### PR TITLE
fix(client-certificates): keep ignoreHTTPSErrors false by default

### PR DESCRIPTION
**Pull request by mxschmitt**
The problem was that we were holding a copy inside `ClientCertificatesInteceptor` but then changing the options [afterwards](https://github.com/microsoft/playwright/blob/6dbc7b54e8ea62c1f31c43aa8dad7e079e136003/packages/playwright-core/src/server/browserContext.ts#L669). Instead of holding a copy, lets only hold the properties we need.

### CODE REVIEW

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 1]
```diff
@@ -98,11 +98,14 @@ class SocksProxyConnection {
...
-    const targetTLS = tls.connect({
+    const tlsOptions: tls.ConnectionOptions = {
       socket: this.target,
-      rejectUnauthorized: this.socksProxy.contextOptions.ignoreHTTPSErrors === true ? false : true,
-      ...clientCertificatesToTLSOptions(this.socksProxy.contextOptions.clientCertificates, `https://${this.host}:${this.port}/`),
-    });
+      rejectUnauthorized: this.socksProxy.ignoreHTTPSErrors === true ? false : true,
+      ...clientCertificatesToTLSOptions(this.socksProxy.clientCertificates, `https://${this.host}:${this.port}/`),
+    };
+    if (process.env.PWTEST_UNSUPPORTED_CUSTOM_CA && isUnderTest())
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:106]: we have a similar hack here https://github.com/microsoft/playwright/blob/6dbc7b54e8ea62c1f31c43aa8dad7e079e136003/packages/playwright-core/src/server/fetch.ts#L199 to pretend to have a real-world site.

File: tests/library/client-certificates.spec.ts [hunk 2]
```diff
@@ -249,6 +249,21 @@ test.describe('browser', () => {
     await page.close();
   });
 
+  test('should not ignoreHTTPSErrors by default', async ({ browser, httpsServer, asset }) => {
```
Comment (mxschmitt) [inline tests/library/client-certificates.spec.ts:252]:
```suggestion
  test('should have ignoreHTTPSErrors=false by default', async ({ browser, httpsServer, asset }) => {
```

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 3]
```diff
@@ -133,10 +136,14 @@ class SocksProxyConnection {
 export class ClientCertificatesProxy {
   _socksProxy: SocksProxy;
   private _connections: Map<string, SocksProxyConnection> = new Map();
+  ignoreHTTPSErrors: boolean | undefined;
+  clientCertificates: { url: string; certs: { cert?: channels.Binary; key?: channels.Binary; passphrase?: string; pfx?: channels.Binary; }[]; }[] | undefined;
```
Comment (dgozman) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:140]:
```suggestion
  clientCertificates: channels.BrowserNewContextOptions['clientCertificates'];
```

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts [hunk 4]
```diff
@@ -98,11 +98,14 @@ class SocksProxyConnection {
     this.internalTLS = internalTLS;
     internalTLS.on('close', () => this.socksProxy._socksProxy.sendSocketEnd({ uid: this.uid }));
 
-    const targetTLS = tls.connect({
+    const tlsOptions: tls.ConnectionOptions = {
       socket: this.target,
-      rejectUnauthorized: this.socksProxy.contextOptions.ignoreHTTPSErrors === true ? false : true,
-      ...clientCertificatesToTLSOptions(this.socksProxy.contextOptions.clientCertificates, `https://${this.host}:${this.port}/`),
-    });
+      rejectUnauthorized: this.socksProxy.ignoreHTTPSErrors === true ? false : true,
```
Comment (yury-s) [inline packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts:103]:
```suggestion
      rejectUnauthorized: !!this.socksProxy.ignoreHTTPSErrors,
```

**Review by yury-s — APPROVED** [review]

**Pull request merged**